cpp/src/db/db_object.cu

```diff
@@ -378,38 +344,34 @@ void db_table<idx_t>::rebuildIndices() {
   for (size_t i = 0; i < columns.size(); i++) {
     // Copy the column's data to a new array
     idx_t size = column_size;
-    idx_t* tempColumn;
-    ALLOC_TRY(&tempColumn, sizeof(idx_t) * size, nullptr);
-    cudaMemcpy(tempColumn, columns[i], sizeof(idx_t) * size, cudaMemcpyDefault);
+    rmm::device_buffer tempColumn(sizeof(idx_t) * size);
+    cudaMemcpy(tempColumn.data(), columns[i].data(), sizeof(idx_t) * size, cudaMemcpyDefault);
 
     // Construct an array of ascending integers
-    idx_t* indirection;
-    ALLOC_TRY(&indirection, sizeof(idx_t) * size, nullptr);
-    thrust::sequence(rmm::exec_policy(nullptr)->on(nullptr), indirection, indirection + size);
+    rmm::device_buffer indirection(sizeof(idx_t) * size);
+    thrust::sequence(rmm::exec_policy(nullptr)->on(nullptr),
+                     (idx_t*) indirection.data(),
+                     (idx_t*) indirection.data() + size);
```

Comment:
C style casting is used extensively in this PR. Better use C++ style casting (e.g. reinterpret_cast here as this code reinterprets void* to idx_t*). And if you don't need to interact with .cpp or .hpp files, rmm::device_vector is generally a safer/cleaner option than rmm::device_buffer (please ignore this comment if you need to interact with .cpp and .hpp files).

Reply:
I have replaced the C-style casting with C++-style casting. Sticking with rmm::device_buffer as it's cleaner to use (no multiple steps to get a raw pointer out).

Reply:
OK, but in general, if you are not using old style functions expecting raw pointers, thrust::device_vector.data() returns device_ptr, device_ptr can be safely dereferenced in both host and device, and this often leads to safer & cleaner implementation in many cases. Not a huge deal, but just a recommendation.